modules/apps/27-interchain-accounts/types/account.go

```diff
@@ -7,17 +7,17 @@ import (
 
 	crypto "github.com/cosmos/cosmos-sdk/crypto/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/address"
 	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
-	"github.com/tendermint/tendermint/crypto/tmhash"
 	yaml "gopkg.in/yaml.v2"
 
 	connectiontypes "github.com/cosmos/ibc-go/v2/modules/core/03-connection/types"
 )
 
-// GenerateAddress returns an sdk.AccAddress using the provided port identifier
-func GenerateAddress(portID string) sdk.AccAddress {
-	return sdk.AccAddress(tmhash.SumTruncated([]byte(portID)))
+// GenerateAddress returns an sdk.AccAddress derived using the provided module account address and port identifier
```

Comment:
Maybe we should mention this is a sub account of the module account?

Reply:
Updated to be more explicit